lightningd/closing_control.c

```diff
@@ -453,6 +454,32 @@ void peer_start_closingd(struct channel *channel,
 		return;
 	}
 
+	// Determine the wallet index for our output, UINT32_MAX if not found.
+	u32 local_wallet_index;
+	struct ext_key local_wallet_ext_key;
+	bool is_p2sh;
+	if (wallet_can_spend(
+		    ld->wallet,
+		    channel->shutdown_scriptpubkey[LOCAL],
+		    &local_wallet_index,
+		    &is_p2sh)) {
+		if (bip32_key_from_parent(
+			    ld->wallet->bip32_base,
+			    local_wallet_index,
+			    BIP32_FLAG_KEY_PUBLIC,
+			    &local_wallet_ext_key) != WALLY_OK) {
+			abort();
```

Comment:
under what circumstances can this fail?

maybe use `channel_internal_error`?

Reply:
fixed